orca/topology/alerts/zabbix/upstream.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2020 OpenRCA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pyzabbix import ZabbixAPI
+from orca.topology import upstream
+from orca.common import config
+
+CONFIG = config.CONFIG
+
+
+class UpstreamProxy(upstream.UpstreamProxy):
+
+    """Upstream proxy for Zabbix."""
+
+    def __init__(self, client):
+        self._client = client
+        self._client.login(CONFIG.probes.zabbix.username, CONFIG.probes.zabbix.password)
+
+    def get_all(self):
+        all =  self._client.trigger.get(
+            only_true=1,
+            active=1,
+            output='extend',
+            selectHosts=['host'])
+        triggers = []
+        for trigger in all:
+            for host in trigger['hosts']:
+                properties = {}
+                properties['host'] = host['host']
+                properties['description'] = trigger['description']
+                properties['status'] = 'active' if trigger['value'] == '1' else 'inactive'
+                properties['priority'] = trigger['priority']
```

Comment:
It would be better to move this logic to the extractor. It is the responsibility of the extractor to determine the `status` based on a `value` and map fields for `priority` and `description`. We shouldn't mix that into the upstream class, which is just a simple abstraction around retrieving payloads from API requests.

Although this may seem counterintuitive, I suggest we reimplement the upstream class to return a "raw" structure of format:

```
[{"host": <host>, "trigger": <trigger>}, ...]
```

Then, process individual items in extractor using the extraction code above.

The `<trigger>` could be a dict of keys [cut from](https://stackoverflow.com/a/5333919) the full trigger payload, limited to only those keys that are required for further extraction.